### PR TITLE
Import of v5 07-ssh content

### PR DESCRIPTION
**Pull request by vuw-ecs-kevin**
Test of my ability to issue a PR by seeding of the renamed SSH file with the old content.

Realisation, post initial PR, that old MD source wasn't compatible with current markup
language syntax.

Since added the other three pages fron the v5 source that appear to have a match
in the new content.

FYI, I also converted the 01-branching, 02-forking, 03-review, 08-exceptions and
09-numbers pages from the v5 content, but there's no obvious place in the new 
shell-extras to put them, but if you want them, just point me to where.

In the absence of further correspondence as to the "seeding" changes, I've now 
added in the editorial changes I'd originally flagged as being less than clear or
where information was missing in the v5 lesson.

Not sure I can actually do anything else?

### CODE REVIEW

**Comment by kynan** [comment]
I think we should follow the convention of only including the `.md` sources in prs and have maintainers regenerate html as needed to avoid spurious merge conflicts.

File: 00-man-pages.md [hunk 1]
```diff
@@ -1,11 +1,141 @@
...
+*   SEE ALSO:
+    list other commands that we might find useful
+    or other sources of information that might help us.
+
+Other sections we might see include
+AUTHOR, REPORTING BUGS, COPYRIGHT, HISTORY, (known) BUGS, and COMPATIBILITY.
+
+#### How to Read the Synopsis
+
+Here is the is synopsis for the `cp` command on Ubuntu Linux:
```
Comment (kynan) [inline 00-man-pages.md:48]: Here is the

**Comment by vuw-ecs-kevin** [comment]
> I think we should follow the convention of only including the `.md`
> sources in prs and have maintainers regenerate html as needed to avoid
> spurious merge conflicts.

Might be some confusion over my understanding of the way the
files that get produced by a "make preview" are stored in the
repository and.or you may not be party to all of the dissussion,
as most of it ocurred, out of band, before I was ready to make
a commit/PR.

I'm from a background (GNU Autotools) where things than can be
"made" usually do not get added into any CVS trcaking, however,
when questioning the versioning of the HTML make targets, I was
told:

> As to the makefile - the current system is to create content and
> submit PRs in the .md files (easier to markup, esp for beginners),
> then the makefile generates the html.  We DO want to version the
> html because when it's committed and pushed to github in the
> gh-pages branch, it then renders nicely as a website (I try to
> update + commit the html pages every few days if there are lots of
> merges).  I haven't committed any of the html for the lesson pages
> since they're all blank right now.  :P But once we've got more
> stuff, there should be a website at
> swcarpentry.github.io/shell-extras.

so I made sure that I "versioned" the HTML generated from the revison
of the MD files I was checking in.

My preference, as discussed with the maintainers, would be to have
a generated file, that gets included into each wbpage, that displays
the Git commit string or even the output from 'git describe' so as
to make it obvious which commit the pages you are viewing come from
without needing to track the HTML to achieve the same end.

However, I'm just doing what I was told to do, and I can see that,
if there "website" content for the repos are (as I was told the
GitHub model does it thing) based in the repo itself then it makes
some sense, because I doubt GitHUb would allow one to upload non-Git
trcaked/versioned files?

I'll happily listen to any other workflow you want me to use though:
I only really had a couple of suggestions for changes to the content
of the old, V5, ssh page, but then saw that, if I was going to put
the old content in for reference, I might as well add the other two
pages that mapped between the v5 and v5.3 source tress.

Maybe I should have waited for the v5.3 repo to "stabilise" before
doing anything

Kevin

**Comment by ChristinaLK** [comment]
This is great!  Thank you so much @vuw-ecs-kevin .  :D  

There's a few things that will need doing (adding entries to a reference.md file, porting over images), but I will post those up as issues - they don't need to be addressed in the PR.  

Typically PR's just have the commit of the .md file and then the maintainer of the repository (someone like me), will generate the html + commit it on a regular basis.  But since there's not much happening in this repo right now, committing the html is fine for this.  

Once again, thanks for doing this!

**Comment by vuw-ecs-kevin** [comment]
> This is great!  Thank you so much @vuw-ecs-kevin .  :D
> 
> There's a few things that will need doing (adding entries to a
> reference.md file, porting over images), but I will post those up as
> issues - they don't need to be addressed in the PR.
> 
> Typically PR's just have the commit of the .md file and then the
> maintainer of the repository (someone like me), will generate the html +
> commit it on a regular basis.  But since there's not much happening in
> this repo right now, committing the html is fine for this.

So should we read that as saying the SVG images sources should be part
of the repo, and hence commit and/or PRs, but the PNGs should not ?

Obviously, I already have the SVG sources for the images from the v5 repo
in my clone of the v5.3 one, and so could easily upload/commit those if
needed and/or it saves you any work, although it's not clear how they
should be named, given that the page names, which is what the old image
names were seemingly based on, have now changed.

Kevin

**Comment by kynan** [comment]
This was intended as a discussion point, not as a critique. The generated html files need to be in the repo so they can be served via GitHub pages because we generate them via Pandoc rather than jekyll.

The pratice I suggested is what we use e.g. in https://github.com/swcarpentry/python-novice-inflammation

**Pull request merged**